src/waltz/xdp/fd_xsk.c

```diff
@@ -448,58 +353,48 @@ fd_xsk_init( fd_xsk_t * xsk ) {
-    return -1;
-  }
-  uint if_idx = if_nametoindex( xsk->if_name_cstr );
-  if( FD_UNLIKELY( if_idx )==0 ) {
-    FD_LOG_WARNING(( "if_nametoindex(%s) failed (%i-%s)", xsk->if_name_cstr, errno, fd_io_strerror( errno ) ));
-    return -1;
+    return NULL;
   }
-  xsk->if_idx = if_idx;
 
   /* Create XDP socket (XSK) */
 
   xsk->xsk_fd = socket( AF_XDP, SOCK_RAW, 0 );
   if( FD_UNLIKELY( xsk->xsk_fd<0 ) ) {
     FD_LOG_WARNING(( "Failed to create XSK (%i-%s)", errno, fd_io_strerror( errno ) ));
-    return -1;
+    return NULL;
   }
 
   /* Associate UMEM region of fd_xsk_t with XSK via setsockopt() */
 
-  if( FD_UNLIKELY( 0!=fd_xsk_setup_umem( xsk ) ) ) return -1;
+  if( FD_UNLIKELY( 0!=fd_xsk_setup_umem( xsk ) ) ) goto fail;
 
   /* Map XSK rings into local address space */
 
-  if( FD_UNLIKELY( 0!=fd_xsk_mmap_ring( &xsk->ring_rx, xsk->xsk_fd, XDP_PGOFF_RX_RING,              sizeof(struct xdp_desc), xsk->params.rx_depth, &xsk->offsets.rx ) ) ) return 0;
-  if( FD_UNLIKELY( 0!=fd_xsk_mmap_ring( &xsk->ring_tx, xsk->xsk_fd, XDP_PGOFF_TX_RING,              sizeof(struct xdp_desc), xsk->params.tx_depth, &xsk->offsets.tx ) ) ) return 0;
-  if( FD_UNLIKELY( 0!=fd_xsk_mmap_ring( &xsk->ring_fr, xsk->xsk_fd, XDP_UMEM_PGOFF_FILL_RING,       sizeof(ulong),           xsk->params.fr_depth, &xsk->offsets.fr ) ) ) return 0;
-  if( FD_UNLIKELY( 0!=fd_xsk_mmap_ring( &xsk->ring_cr, xsk->xsk_fd, XDP_UMEM_PGOFF_COMPLETION_RING, sizeof(ulong),           xsk->params.cr_depth, &xsk->offsets.cr ) ) ) return 0;
+  if( FD_UNLIKELY( 0!=fd_xsk_mmap_ring( &xsk->ring_rx, xsk->xsk_fd, XDP_PGOFF_RX_RING,              sizeof(struct xdp_desc), xsk->params.rx_depth, &xsk->offsets.rx ) ) ) goto fail;
+  if( FD_UNLIKELY( 0!=fd_xsk_mmap_ring( &xsk->ring_tx, xsk->xsk_fd, XDP_PGOFF_TX_RING,              sizeof(struct xdp_desc), xsk->params.tx_depth, &xsk->offsets.tx ) ) ) goto fail;
+  if( FD_UNLIKELY( 0!=fd_xsk_mmap_ring( &xsk->ring_fr, xsk->xsk_fd, XDP_UMEM_PGOFF_FILL_RING,       sizeof(ulong),           xsk->params.fr_depth, &xsk->offsets.fr ) ) ) goto fail;
+  if( FD_UNLIKELY( 0!=fd_xsk_mmap_ring( &xsk->ring_cr, xsk->xsk_fd, XDP_UMEM_PGOFF_COMPLETION_RING, sizeof(ulong),           xsk->params.cr_depth, &xsk->offsets.cr ) ) ) goto fail;
 
   /* Bind XSK to queue on network interface */
 
   struct sockaddr_xdp sa = {
     .sxdp_family   = PF_XDP,
-    .sxdp_ifindex  = xsk->if_idx,
-    .sxdp_queue_id = xsk->if_queue_id,
+    .sxdp_ifindex  = if_idx,
+    .sxdp_queue_id = if_queue,
     /* See extended commentary below for details on on
        XDP_USE_NEED_WAKEUP flag. */
     .sxdp_flags    = XDP_USE_NEED_WAKEUP | (ushort)xsk->params.zerocopy
   };
 
   if( FD_UNLIKELY( 0!=bind( xsk->xsk_fd, (void *)&sa, sizeof(struct sockaddr_xdp) ) ) ) {
-    FD_LOG_WARNING(( "Unable to bind to interface %s queue %u (%i-%s)",
-                     xsk->if_name_cstr, xsk->if_queue_id, errno, fd_io_strerror( errno ) ));
-    return -1;
+    char if_name[ IF_NAMESIZE+1 ] = {0};
```

Comment:
```suggestion
    char if_name[ IF_NAMESIZE ] = {0};
```
Does this include the null terminator already?